library/objective-c/EnvoyEngine.h

```diff
@@ -82,7 +62,7 @@ NS_ASSUME_NONNULL_BEGIN
  @param config The configuration file with which to start Envoy.
  @return A status indicating if the action was successful.
  */
-+ (EnvoyStatus)runWithConfig:(NSString *)config;
++ (envoy_engine_t)runWithConfig:(NSString *)config;
```

Comment:
Let's move these functions to a new `@protocol Engine`, then have `@interface EnvoyEngine` implement them, as discussed offline so that we can mock these from the platform. This aligns with what Android has on master

Reply:
Are there potential namespacing issues there? We could do `EnvoyEngineProtocol`..

Reply:
Yea, as we discussed offline, let's go with `EnvoyEngine` as the protocol and `EnvoyEngineImpl` as the class. We should also update this on the Java side as part of this PR if possible.
Just to document here, the "impl" suffix naming is a convention used in Envoy, which is why we're opting to use it in Objc/Java (but not Kotlin/Swift)